pkg/providers/imagefamily/nodeimage_test.go

```diff
@@ -171,39 +200,101 @@ var _ = Describe("NodeImageProvider tests", func() {
+		})
+
 		// This test changes depending on the Kubernetes version, in effect making the following version-specific tests unnecessary.
 		// They are still kept for clarity and to ensure that the behavior is explicitly tested.
-		It("should match expected images for AzureLinux, depending on the Kubernetes version", func() {
+		It("should match expected images for default AzureLinux, depending on the Kubernetes version, when FIPSMode is explicitly set to Disabled", func() {
 			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr("Disabled")
 
 			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
 		})
 
-		It("should match expected images for AzureLinux with version < 1.32", func() {
+		// This test changes depending on the Kubernetes version, in effect making the following version-specific tests unnecessary.
+		// They are still kept for clarity and to ensure that the behavior is explicitly tested.
+		It("should match expected images for default AzureLinux, depending on the Kubernetes version, when FIPSMode is not explicitly set", func() {
 			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
-			nodeClass.Status.KubernetesVersion = "1.31.0"
 
 			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, "1.31.0")))
+			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
 		})
 
-		It("should match expected images for AzureLinux with version >= 1.32", func() {
+		// This test changes depending on the Kubernetes version, in effect making the following version-specific tests unnecessary.
+		// They are still kept for clarity and to ensure that the behavior is explicitly tested.
+		It("should match expected images for FIPS AzureLinux, depending on the Kubernetes version, when FIPSMode is explicitly set to FIPS", func() {
 			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
-			nodeClass.Status.KubernetesVersion = "1.32.0"
+			nodeClass.Spec.FIPSMode = lo.ToPtr("FIPS")
 
 			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, "1.32.0")))
+			Expect(foundImages).To(ContainElements(getExpectedTestFIPSImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
 		})
+
+		DescribeTable("should match expected images",
+			func(imageFamily *string, fipsMode *string, version string, kubernetesVersion string) {
+				nodeClass.Spec.ImageFamily = imageFamily
+				effectiveFipsMode := ""
```

Comment:
I think it would be a bit simpler if you just have `fipsMode` be a string, instead of a `*string`.

Than you don't need `effectiveFipsMode`, and can do your if check on `fipsMode != nil` as `fipsMode != ""` instead.